packages/engine/lib/orchestrator/src/process/local.rs

```diff
@@ -29,17 +29,67 @@ pub struct LocalProcess {
 
 #[async_trait]
 impl process::Process for LocalProcess {
-    async fn exit_and_cleanup(mut self: Box<Self>) -> Result<()> {
-        self.child
-            .kill()
-            .await
-            .or_else(|e| match e.kind() {
-                // From `Child::kill` docs: Forces the child process to exit. If the child has
-                // already exited, an InvalidInput error is returned
-                std::io::ErrorKind::InvalidInput => Ok(()),
-                _ => Err(Report::new(e)),
-            })
-            .wrap_err("Could not kill the process")?;
+    async fn exit_and_cleanup(mut self: Box<Self>, experiment_id: ExperimentId) -> Result<()> {
+        // Leave some time for the engine to cleanup some resources
+        if let Err(_) =
+            tokio::time::timeout(tokio::time::Duration::from_millis(500), self.child.wait()).await
+        {
+            // Kill the child process as it didn't stop on its own
+            self.child
+                .kill()
+                .await
+                .or_else(|e| match e.kind() {
+                    // From `Child::kill` docs: Forces the child process to exit. If the child has
+                    // already exited, an InvalidInput error is returned
+                    std::io::ErrorKind::InvalidInput => Ok(()),
+                    _ => Err(Report::new(e)),
+                })
+                .wrap_err("Could not kill the process")?;
+        }
+
+        // Cleanup python socket files in case the engine didn't
+        if let Ok(current_dir) = std::env::current_dir() {
```

Comment:
`cleanup_experiment` should do exactly what the name suggests: it cleans up the things the specified experiment did. We really should use that method instead of reinventing the wheel here.